File: python/Services/ViaCEP.py
from Contracts.CepProviderInterface import CepProviderInterface
import requests


class ViaCEP(CepProviderInterface):
    def search(self, cep: str) -> dict:
        response = requests.get(f"https://viacep.com.br/ws/{cep}/json/")

        data = response.json()

        return {
            "cep": data.get("cep"),
            "rua": data.get("logradouro"),
            "bairro": data.get("bairro"),
            "cidade": data.get("localidade"),
            "fonte": "ViaCEP"
        }

File: php/src/Services/ViaCEP.php
<?php

namespace php\Services;

use php\Contracts\CepProviderInterface;

class ViaCEP implements CepProviderInterface
{
    public function search(string $cep): array
    {
        $response = file_get_contents("https://viacep.com.br/ws/{$cep}/json/");

        $data = json_decode($response, true);

        return [
            'cep' => $data['cep'] ?? '',
            'rua' => $data['logradouro'] ?? '',
            'bairro' => $data['bairro'] ?? '',
            'cidade' => $data['localidade'] ?? '',
            'fonte' => 'ViaCEP'
        ];
    }
}
